pkg/apis/serving/v1beta1/route_types.go

```diff
@@ -89,10 +89,17 @@ type TrafficTarget struct {
 	// +optional
 	LatestRevision *bool `json:"latestRevision,omitempty"`
 
-	// Percent specifies percent of the traffic to this Revision or Configuration.
-	// This defaults to zero if unspecified.
+	// Percent indicates that percentage based routing should be used and
+	// the value indicates the percent of traffic that is be routed to this
+	// Revision or Configuration. `0` (zero) mean no traffic, `100` means all
+	// traffic.
+	// If "percent" appears on one or more Traffic Targets then the sum of
```

Comment:
We should wait to update this until we have new routing mechanisms. The change to a pointer gives us increased flexibility in building new routing modes into Knative, but Lines 96-100 are both confusing and incorrect with today's behavior.

Reply:
As I mention here: https://github.com/knative/serving/pull/4947#issuecomment-517294061
I don't think the current wording allows for an extension routing mechanisms to be written today and we'd like to start to create proprietary ones, and use them, today w/o waiting for the community to adopt a new one but still not break existing expectations on how to interpret the percent flag.

If you think 96-100 is confusing please suggest alternative text, but I don't see what's incorrect - please elaborate.

Reply:
I tweaked the wording a bit - see if that helps